diff --git a/Headers/Foundation/NSZone.h b/Headers/Foundation/NSZone.h
index e0e7ad84c..e7ac7caa1 100644
--- a/Headers/Foundation/NSZone.h
+++ b/Headers/Foundation/NSZone.h
@@ -247,14 +247,6 @@ NSReallocateCollectable(void *ptr, NSUInteger size, NSUInte
ger options);                                                                    
 #endif
 
-static inline id NSMakeCollectable(const void *cf) {
-#if __has_feature(objc_arc)
-    return nil;
-#else
-    return (id)cf; // Unimplemented; garbage collection is deprecated.
-#endif
-}
-
 #if	defined(__cplusplus)
 }
 #endif
